crates/miden-tx/src/host/account_procedures.rs

```diff
@@ -38,6 +41,22 @@ impl AccountProcedureIndexMap {
         Ok(Self(result))
     }
 
+    /// Builds an [`AccountProcedureIndexMap`] for the specified transaction inputs and arguments.
+    ///
+    /// The resulting instance will map all account code commmitments to a mapping of
+    /// `proc_root |-> proc_index` for any account that is expected to be involved in the
+    /// transaction, enabling easy procedure index lookups on runtime.
+    pub fn from_transaction_params(
+        tx_inputs: &TransactionInputs,
+        tx_args: &TransactionArgs,
+        advice_inputs: &TransactionAdviceInputs,
+    ) -> Result<Self, TransactionHostError> {
+        let mut account_code_commitments = tx_args.foreign_account_code_commitments();
+        account_code_commitments.insert(tx_inputs.account().code().commitment());
```

Comment:
Nit: I found it a bit surprising that `tx_args.foreign_account_code_commitments()` returned an owned value rather than being a getter. I would have expected it to be prefixed with `to_` (https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv) - should we change that?